File: spring-boot/spring-boot-testing/base-java-h2/src/main/java/com/sesample/notes/entities/Customer.java
package com.sesample.notes.entities;

import javax.persistence.*;

/**
 * @author Evgeniy Skiba on 14.11.2020
 * @project base-java-h2
 */

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private final String name;

    @Column(nullable = false)
    private final String email;

    public Customer() {
        this(null,null);
    }

    public Customer(String name, String email) {
        this.name = name;
        this.email = email;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    // Other code omitted
}

File: old/SpringBoot/springboot-kotlin-h2-sample/src/main/kotlin/com/se/example/springbootkotlinh2sample/entities/Customer.kt
package com.se.example.springbootkotlinh2sample.entities

import javax.annotation.Generated
import javax.persistence.*

@Entity
class Customer(

        @EmbeddedId
        var customerId: CustomerId? = null,

        @Column(name = "company")
        var company: String = "",
        var name: String
)